server/manager/src/main/java/org/apache/accumulo/manager/metrics/fate/FateMetrics.java

```diff
@@ -29,12 +29,17 @@
 import org.apache.accumulo.fate.ZooStore;
 import org.apache.accumulo.manager.metrics.ManagerMetrics;
 import org.apache.accumulo.server.ServerContext;
-import org.apache.hadoop.metrics2.lib.MetricsRegistry;
-import org.apache.hadoop.metrics2.lib.MutableGaugeLong;
+// import org.apache.hadoop.metrics2.lib.MetricsRegistry;
+// import org.apache.hadoop.metrics2.lib.MutableGaugeLong;
 import org.apache.zookeeper.KeeperException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
```

Comment:
For an example, this may be okay - but for production I think I'd expect either a CompositeMeterRegisry that was created / managed by the metrics context - or would use the Metrics.globalRegisrty.

Reply:
I agree. This is one of the things I was hoping to get input on. Both seem like they could work however I'm not sure which would be best.

Reply:
With the current initialization occurring in a base class, the composite seems the likely candidate, but global may be just as well - it may depend on / if there are multiple tenant situations - say in a cluster all things run in a separate process, then there would be no conflicts, but if say in mini-cluster (or another situation) where the registry was shared by multiple "processes" then there may need to be a way to differentiate between them - so that if there were metrics named the same, but for different processes they would not step on each other.

With the global registry, this could be mitigated by include a process name tag (or something)

In general, a process tag might already be baked into the current naming scheme so it would be a non-issue just to carry that forward.  There may be guidance in naming conventions and what makes a suitable tag.